src/main/java/io/spaship/operator/service/k8s/Operator.java

```diff
@@ -37,414 +36,413 @@
+                eb.websiteName(item.getMetadata().getLabels().get(WEBSITE))
+                        .environmentName(item.getMetadata().getLabels().get(ENVIRONMENT))
+                        .state("deployment created");
+            }
+            if (item instanceof StatefulSet sfs) {
+                LOG.debug("creating new Deployment in K8s, tracing = {}", tracing);
+                ocClient.apps().statefulSets().inNamespace(nameSpace).createOrReplace(sfs);
+                eb.websiteName(item.getMetadata().getLabels().get(WEBSITE))
+                        .environmentName(item.getMetadata().getLabels().get(ENVIRONMENT))
+                        .state("StatefulSet created");
+
+            }
+            if (item instanceof PersistentVolumeClaim pvc) {
+                LOG.debug("creating new pvc in K8s, tracing = {}", tracing);
+                ocClient.persistentVolumeClaims().inNamespace(nameSpace).createOrReplace(pvc);
+                eb.websiteName(item.getMetadata().getLabels().get(WEBSITE))
+                        .environmentName(item.getMetadata().getLabels().get(ENVIRONMENT))
+                        .state("pvc created");
+            }
+            if (item instanceof Route route) {
+                LOG.debug("creating new Route in K8s, tracing = {}", tracing);
+                ocClient.routes().inNamespace(nameSpace).createOrReplace(route);
+                eb.websiteName(item.getMetadata().getLabels().get(WEBSITE))
+                        .environmentName(item.getMetadata().getLabels().get(ENVIRONMENT))
+                        .state("route created");
+            }
+            if (item instanceof ConfigMap cm) {
+                LOG.debug("creating new ConfigMap in K8s, tracing = {}", tracing);
+                ocClient.configMaps().inNamespace(nameSpace).createOrReplace(cm);
+                eb.websiteName(item.getMetadata().getLabels().get(WEBSITE))
+                        .environmentName(item.getMetadata().getLabels().get(ENVIRONMENT))
+                        .state("configmap created");
+            }
+            if (item instanceof Ingress ing) {
+                LOG.debug("creating new Ingress controller in K8s, tracing = {}", tracing);
+                ocClient.network().v1().ingresses().inNamespace(nameSpace).createOrReplace(ing);
+            }
+
+            eventManager.queue(eb.build());
+        });
+
+    }
+
+    public ConfigMap updateConfigMap(Environment environment, Object syncConfig) {
+
+        var configMapName = "sidecar-config-"
+                .concat(environment.getWebsiteName().toLowerCase())
+                .concat("-")
+                .concat(environment.getName().toLowerCase());
+        var cfgMapData = ocClient.configMaps().inNamespace(environment.getNameSpace()).withName(configMapName).get();
+        var existingConfigData = cfgMapData.getData();
+        existingConfigData.put("SIDECAR_SYNC_CONFIG", syncConfig.toString());
```

Comment:
can we check `cfgMapData` before using it? that if data exists or not as we're getting it from `ocClient`